src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,637 @@
+						};
+					});
+					this.logger.warning(`Ignored timetable entries from WebUntis import`, { ignored: filteredTimetable });
+				}
+
+				timetable = timetable.filter(entry => entry.te.length === 1
+					&& entry.te[0].id === teacher.id
+					&& entry.kl.length > 0
+					&& entry.ro.length === 1
+					&& entry.su.length === 1);
+				/* END TODO: Check for change */
+
+				const compareArraysOfString = (a1, a2) => {
+					if (a1.length !== a2.length) {
+						return false;
+					}
+
+					for (let i = 0; i < a1.length; ++i) {
+						if (a1[i] !== a2[i]) {
+							return false;
+						}
+					}
+
+					return true;
+				};
+	
+				for (const entry of timetable) {
+					const classNames = entry.kl.map(k => k.longname);
+
+					// Select course with matching teacher, subject, and participating classes
+					let course = intermediateData.courses.find(course => {
+						return course.teacher === currentTeacherName &&
+							course.subject === entry.su[0].longname &&
+							compareArraysOfString(course.classes, classNames);
+					});
+
+					if (course === undefined) { // Create course
+						course = {
+							teacher: currentTeacherName,
+							subject: entry.su[0].longname,
+							classes: classNames,
+							timetable: []
+						};
+						intermediateData.courses.push(course);
+					}
+					
+					// Update timetable of course
+					course.timetable.push({
+						date: entry.date,
+						startTime: entry.startTime,
+						endTime: entry.endTime,
+						room: entry.ro[0].longname,
+					});
+
+					// Update first and last date of course
+					if (course.startDate === undefined || course.startDate > entry.date) {
+						course.startDate = entry.date;
+					}
+					if (course.endDate === undefined || course.endDate < entry.date) {
+						course.endDate = entry.date;
+					}
+				}
+			}
+		} else if (intermediateData.rooms !== undefined) { // Iterate over rooms
+			for (const room of intermediateData.rooms) {
+				const currentRoomName = `${room.name} (${room.longName}${room.building !== '' ? `, ${room.building}` : ''})`;
+
+				let timetable = await this.getCustomizableTimeTableFor(4, room.id, {
+					startDate: intermediateData.currentSchoolYear.startDate,
+					endDate: intermediateData.currentSchoolYear.endDate,
+					onlyBaseTimetable: true,
+					klasseFields: ['id', 'longname'],
+					subjectFields: ['id', 'longname'],
+					teacherFields: ['id', 'longname'],
+					roomFields: ['id', 'longname'],
+				});
+
+				/* TODO: Check for change */
+				let filteredTimetable = timetable.filter(entry => !(entry.ro.length === 1
+					&& entry.ro[0].id === room.id
+					&& entry.kl.length > 0
+					&& entry.te.length === 1
+					&& entry.su.length === 1));
+				if (filteredTimetable.length > 0) {
+					filteredTimetable = filteredTimetable.map(entry => {
+						return {
+							teachers: entry.te.map(teacher => teacher.longname),
+							teacherValid: entry.te.length === 1,
+							rooms: entry.ro.map(room => room.longname),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['room' is already declared in the upper scope.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)